scripts/helpers.js

```diff
@@ -18,6 +19,40 @@ try {
 } catch (e) {}
 
 /**
+ * Generate data structure for generating table of contents.
+ *
+ * @param {string} content - Page content.
+ * @returns {array} [{title: 'Title', link: '#anchor', children: []}]
+ */
+hexo.extend.helper.register('table_of_contents', function (content) {
+  var $ = cheerio.load(content);
+  var h2Set = $('h2');
+
+  // Get H2s.
+  return h2Set.map(function (h2) {
+    var $h2 = $(this);
+
+    // Add H2.
+    var link = $h2.find('a')[0];
```

Comment:
could do

```js
var $link = $h2.find('a').eq(0);
var item = {
  title: $link.attr('title'),
  href: $link.attr('href'),
  children: []
};
```

it's obviously not any faster, but it's consistent. prob easiest to keep what you have though.